tests/test_serving.py

```diff
@@ -278,3 +278,27 @@ def app(environ, start_response):
         'content_type': 'application/json'
     })
     assert r.content == b'YES'
+
+
+def test_port_must_be_integer(dev_server):
+    with pytest.raises(TypeError) as port_type_exception:
```

Comment:
Unfortunately this sort of test doesn't work because the server will be started in an entirely different process. The TypeError does happen, but you won't catch it here. Instead I think you should call `run_simple` directly.